src/index.js

```diff
@@ -549,9 +550,15 @@ function createConfig(options, entry, format, writeMeta) {
 									...(options.generateTypes !== false && {
 										declarationDir: getDeclarationDir({ options, pkg }),
 									}),
-									jsx: 'preserve',
-									jsxFactory: options.jsx,
-									jsxFragmentFactory: options.jsxFragment,
+									...(!options.jsxImportSource && {
+										jsx: 'preserve',
+										jsxFactory: options.jsx,
+										jsxFragmentFactory: options.jsxFragment,
+									}),
+									...(options.jsxImportSource && {
+										jsx: 'react-jsx',
+										jsxImportSource: options.jsxImportSource,
+									}),
```

Comment:
Theoretically we should be fine to leave `jsx` untouched & remove that second conditional properties block, but that relies on the assumption that a user has a `tsconfig.json` to use.

Conditionally adding `jsxFactory` and `jsxFragmentFactory` is enough to fix the errors, but somehow `jsx` will need to become `'react-jsx'` and `jsxImportSource` will need to be set to result in valid typings. While users will likely have a correct `tsconfig.json` that will get merged over our defaults, if they don't, the types will just contain a bunch of `any`.

Setting these ourselves allows us to make fewer assumptions about the user's configs that are external to microbundle.